python/lsst/ip/isr/measureCrosstalk.py

```diff
@@ -316,17 +358,193 @@ def reduce(self, ratioList):
             ratios[ii][jj] = result
-        coeff, coeffErr, coeffNum = measureCrosstalkCoefficients(ratios, self.config.rejIter,
-                                                                 self.config.rejSigma)
+            self.debugRatios('reduce', ratios, ii, jj)
+        coeff, coeffErr, coeffNum = self.measureCrosstalkCoefficients(ratios, self.config.rejIter,
+                                                                      self.config.rejSigma)
         self.log.info("Coefficients:\n%s\n", coeff)
         self.log.info("Errors:\n%s\n", coeffErr)
         self.log.info("Numbers:\n%s\n", coeffNum)
         return coeff, coeffErr, coeffNum
 
-    def _getConfigName(self):
-        """Disable config output"""
-        return None
+    def measureCrosstalkCoefficients(self, ratios, rejIter=3, rejSigma=2.0):
+        """Measure crosstalk coefficients from the ratios.
 
-    def _getMetadataName(self):
-        """Disable metdata output"""
-        return None
+        Given a list of ratios for each target/source amp combination,
+        we measure a sigma clipped mean and error.
+
+        The coefficient errors returned are the standard deviation of
+        the final set of clipped input ratios.
+
+        Parameters
+        ----------
+        ratios : `list` of `list` of `numpy.ndarray`
+           Matrix of arrays of ratios.
+        rejIter : `int`
+           Number of rejection iterations.
+        rejSigma : `float`
+           Rejection threshold (sigma).
+
+        Returns
+        -------
+        coeff : `numpy.ndarray`
+            Crosstalk coefficients.
+        coeffErr : `numpy.ndarray`
+            Crosstalk coefficient errors.
+        coeffNum : `numpy.ndarray`
+            Number of pixels for each measurement.
+
+        Notes
+        -----
+        This has been moved into MeasureCrosstalkTask to allow for easier
+        debugging.
+
+        The lsstDebug.Info() method can be rewritten for __name__ =
+        `lsst.ip.isr.measureCrosstalk`, and supports the parameters:
+
+        debug.display['measure'] : `bool`
+            Display a histogram of the combined ratio measurements for
+            a pair of source/target amplifiers from the final set of
+            clipped input ratios.
+        """
+        if rejIter is None:
+            rejIter = self.config.rejIter
+        if rejSigma is None:
+            rejSigma = self.config.rejSigma
+
+        numAmps = len(ratios)
+        assert all(len(rr) == numAmps for rr in ratios)
+
+        coeff = np.zeros((numAmps, numAmps))
+        coeffErr = np.zeros((numAmps, numAmps))
+        coeffNum = np.zeros((numAmps, numAmps), dtype=int)
+
+        for ii, jj in itertools.product(range(numAmps), range(numAmps)):
+            if ii == jj:
+                values = [0.0]
+            else:
+                values = np.array(ratios[ii][jj])
+                values = values[np.abs(values) < 1.0]  # Discard unreasonable values
```

Comment:
Interesting. I wonder if this is worth logging when it happens. Maybe not, just a thought. Also, while 1 is an obvious amount, I wonder if it might be better to have it as a config option? Again, feel free to ignore.

Reply:
I'm ignoring because I think this would be a case that could come up a lot.  If a bright star in amp A ends up on an even brighter star in amp B, we want to kick out that sample from amp A.